modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/showSidebar.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from 'frontend-js-react-web';
+
+import SidebarPanel from '../SidebarPanel';
+import SidebarPanelInfoView from '../components/SidebarPanelInfoView';
+import SidebarPanelMetricsView from '../components/SidebarPanelMetricsView';
+
+const actions = {
+	showInfo(fetchURL, portletNamespace) {
+		showSidebar({View: SidebarPanelInfoView, fetchURL, portletNamespace});
+	},
+	showMetrics(fetchURL, portletNamespace) {
+		showSidebar({
+			View: SidebarPanelMetricsView,
+			fetchURL,
+			portletNamespace,
+		});
+	},
+};
+
+const showSidebar = ({View, fetchURL, portletNamespace}) => {
+	const id = `${portletNamespace}sidebar`;
+
+	const sidebarPanel = Liferay.component(id);
+
+	if (!sidebarPanel) {
+		const container = document.body.appendChild(
+			document.createElement('div')
+		);
+
+		render(
+			SidebarPanel,
+			{
+				fetchURL,
+				ref: (element) => {
+					Liferay.component(id, element);
+				},
+				viewComponent: View,
+			},
+			container
+		);
+	}
+	else {
+		sidebarPanel.open(fetchURL, View);
+	}
+};
+
+export {actions};
+export default showSidebar;
```

Comment:
Are you using `showSidebar` anywhere else? The less API we expose the better... especially as we grow more and more conscious about breaking changes and public APIs.

I know you've already tweaked this with @wincent, but it still feels weird to read:

```
import {actions} from './utils/showSidebar'
```

When there's no visible `import showSidebar from './utils/showSidebar'`. It might be somewhere else, I just haven't checked, but since we've grouped everything in the `clay:dropdown-actions` tag now, it might be no longer necessary?

Reply:
So, to me, reading this, it feels like the default export here is `actions`, so maybe calling this `dropdownActions.js` would make more sense overall? I don't see any qualifier on the dropdown so that we could more easily say something like `contentDashboardItemDropdonActions`, but maybe we only have 1 type of dropdown here anyways?

Reply:
Looking further... is it possible we extracted this so that we could use it from 2 different components but this is no longer needed so we can just put it back inside the `propsTransformer` and rejoice?

Reply:
Yes, I can revert it and place it in the transformer again

Reply:
It's okay, I was just rambling... this might be more organized, and as long as we don't export the APIs as part of the package `main` entry, it should not be considered public API, so whatever makes more sense for readability, debuggability and maintainability to you and the team.

Reply:
I didn't like at all... a util called `showSidebar` that doesn't export `showSidebar` but `actions`... meh, I think it's more logic having it in the transformer now that the code it's not shared :)

Reply:
![done-well](https://user-images.githubusercontent.com/7074/88927370-96baff00-d277-11ea-9f1f-bbdbfdccf3c9.gif)